staging/src/k8s.io/kubectl/pkg/cmd/version/version.go

```diff
@@ -80,6 +81,13 @@ func NewCmdVersion(f cmdutil.Factory, ioStreams genericiooptions.IOStreams) *cob
 		Long:    i18n.T("Print the client and server version information for the current context."),
 		Example: versionExample,
 		Run: func(cmd *cobra.Command, args []string) {
+			warningsAsErrors := false
+			if cmd.Parent() != nil {
+				if flag := cmd.Parent().PersistentFlags().Lookup("warnings-as-errors"); flag != nil {
```

Comment:
```golang
cmd.Flags().Lookup("warnings-as-errors")
```
will be sufficient to read that flag. With that change you should be able to move this entire block to `Complete` function, to match how we're completing the options struct.